pkg/fatfs/fatfs_vfs/fatfs_vfs.c

```diff
@@ -90,6 +90,9 @@ static int _format(vfs_mount_t *mountp)
 
     /* make sure the volume has been initialized */
     if (_init(mountp)) {
+        if (!IS_USED(CONFIG_FATFS_FORMAT_ALLOC_STATIC)) {
```

Comment:
LLVM performs static analysis prior to dead code elimination. In thr case that `IS_USED(CONFIG_FATFS_FORMAT_ALLOC_STATIC)` the `free()` is never reached. Still, LLVM detects that this is a semantic error that `free()` would be called (in a dead branch) on an object that is not allocated on the heap.

Even though I'm personally not a fan of using the C preprocessor when other options are available: I think the `free()` here has to be guarded by a preprocessor conditional to work with LLVM.